core/src/data/literal.rs

```diff
@@ -261,13 +269,58 @@ impl<'a> Literal<'a> {
 
 #[cfg(test)]
 mod tests {
-
     use {
         super::Literal::*,
         bigdecimal::BigDecimal,
         std::{borrow::Cow, str::FromStr},
     };
 
+    #[test]
+    fn try_from_ast_literal() {
+        use {
+            super::{Literal, LiteralError},
+            crate::{
+                ast::{AstLiteral, DateTimeField},
+                data::Interval as I,
+                result::Result,
+            },
+        };
+
+        fn test(ast_literal: AstLiteral, literal: Result<Literal>) {
```

Comment:
Is there any reason not to use closures?

Reply:
closure has two roles in Rust.
- captures outer environment
- allows omitting types

Here, compiler requires the function to specify types explicitly, so one of the pros was not for this.
And here, `test` does not capture any outer variables, so I thought it would be better to simply use normal fn, not closure.